### PR TITLE
Add missing function return types (mostly None) to reloader.py

### PR DESCRIPTION
**Pull request by dave42w**
## Description

This PR improves type hinting by adding return type hints to the functions in reloader.py that didn't have them.

## PR Checklist

Please ensure that:

- [X ] The PR contains a descriptive title
- [X ] The PR contains a descriptive summary of the changes
- [X ] You build and test your changes before submitting a PR.
- [X ] You have added relevant documentation
- [X ] You have added relevant tests. We prefer integration tests wherever possible

## Pre-Commit Instructions:

- [X ] Ensure that you have run the [pre-commit hooks](https://github.com/sparckles/robyn#%EF%B8%8F-to-develop-locally) on your PR.

### CODE REVIEW

File: robyn/reloader.py [hunk 1]
```diff
@@ -117,16 +115,16 @@ class EventHandler(FileSystemEventHandler):
     def __init__(self, file_path: str, directory_path: str) -> None:
         self.file_path = file_path
         self.directory_path = directory_path
-        self.process = None  # Keep track of the subprocess
+        self.process: subprocess.Popen[bytes] | None = None  # Keep track of the subprocess
```
Comment (sansyrox) [inline robyn/reloader.py:118]: Hey @dave42w 👋 

Can you use `Union` here instead? We still have a python 3.8 ci right now. Yes, we should get rid of it soon but we have it running right now 😓

**Comment by dave42w** [comment]
Hi @sansyrox 
I'm getting lots of false ci failures at the moment (mostly timeouts). Is that upgrade to uv going to fix these?

File: robyn/reloader.py [hunk 2]
```diff
@@ -4,17 +4,15 @@
...
 from watchdog.events import FileSystemEventHandler
 from watchdog.observers import Observer
 
 from robyn.logger import Colors, logger
 
-dir_path = None
 
-
-def compile_rust_files(directory_path: str):
+def compile_rust_files(directory_path: str) -> list[str]:
```
Comment (sansyrox) [inline robyn/reloader.py:15]: Hey @dave42w 👋 

This is not a timeout 😅 
```suggestion
def compile_rust_files(directory_path: str) -> List[str]:
```

**Review by sansyrox — COMMENTED** [review]
Hey @dave42w 👋 

Thank you for the PR, just have one suggestion

File: robyn/reloader.py (hunk 2 above)
Comment (dave42w) [reply robyn/reloader.py]: Oops fixed

**Comment by dave42w** [comment]
> Hey @dave42w 👋 
> 
> Thank you for the PR, just have one suggestion

Sorry about that.

I'm getting so many ci errors that it's hard to spot the real ones.
Also a worry that some real errors only show up on only one platform when it should be consistent. 
Anyway this is now showing the rust warning.
https://github.com/sparckles/Robyn/actions/runs/12095320152/job/33728364712#step:4:1

**Review by sansyrox — COMMENTED** [review]
LGTM! 🚀

**Pull request merged**